documents/design/oavb_ipc.asciidoc

```diff
@@ -0,0 +1,83 @@
+= Interprocess Communication Recommendations for AVB/TSN
+Andrew Elder
+v0.1
+
+== Introduction
+
+The AVB/TSN software stack is comprised of a number of modules that implement protocol stacks specified by IEEE standards. Each of these protocols implement an Ethernet network facing interface and an application facing interface. The required application facing interface is described in some detail in http://avnu.org/wp-content/uploads/2014/05/AVnu_SWAPIs_v1.0.pdf[AVB Software Interfaces and Endpoint Architecture Guidelines]. Since a module running a particular protocol may be a standalone execution unit (for example a Linux daemon), an interprocess communication method is required to interface to the execution unit.
+
+For portability reasons it is recommended to separate the implementation of the interprocess communication method as much as possible from the logic of the protocol implementation. This applies to both an application that is interfacing to an AVB/TSN module and the implementation of the AVB/TSN module itself.
+
+For the purposes of this document the following straw-man application interfaces are described:
+
+.Straw-man Interfaces
+[source,c/c++]
+----
+
+void AVBTSN_SetParam(uint32_t param);
+
+----
+
+where the AVBTSN portion of the function name would be replaced by the name of the IEEE AVB/TSN protocol module that is being targeted.
```

Comment:
Thanks for the feedback. In my opinion a "set" of parameters would be communicated by adding multiple parameters to the parameter list and then bundling them internally in a structure. I'll add an example showing that since it is obviously not clear at this stage. I intend datatypes to vary on a parameter by parameter basis.

Hmmm - have to this about error handling. Ideas?

Can you elaborate on the hierarchy of interface identifiers idea? I was thinking each functional operation would map internally to a unique ID in the packing/unpacking layer that would support correct handling of the transaction, but I'm not sure if that is what you are referring you. I need to add more text. Trouble is it is hard to document how it might work without generating a concrete example.